modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/provider/ObjectEntryInfoItemFormProvider.java

```diff
@@ -614,6 +622,52 @@ private InfoFieldSet _getObjectDefinitionInfoFieldSet(
 		).build();
 	}
 
+	private List<InfoFieldSetEntry> _getObjectRelationshipsInfoFieldSets(
+			long objectDefinitionId)
+		throws NoSuchFormVariationException {
+
+		List<InfoFieldSetEntry> infoFieldSetEntries = new ArrayList<>();
+
+		if (objectDefinitionId == 0) {
+			return infoFieldSetEntries;
+		}
+
+		List<ObjectRelationship> objectRelationships =
+			_objectRelationshipLocalService.getAllObjectRelationships(
+				objectDefinitionId);
+
+		for (ObjectRelationship objectRelationship : objectRelationships) {
+			if (!Objects.equals(
+					objectRelationship.getType(),
+					ObjectRelationshipConstants.TYPE_ONE_TO_MANY)) {
+
+				continue;
+			}
+
+			long objectDefinitionId1 =
+				objectRelationship.getObjectDefinitionId1();
+
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.fetchObjectDefinition(
+					objectDefinitionId1);
+
+			if (objectDefinition == null) {
+				throw new NoSuchFormVariationException(
+					String.valueOf(objectDefinitionId1),
+					new NoSuchObjectDefinitionException());
+			}
+
+			infoFieldSetEntries.add(
+				_getObjectDefinitionInfoFieldSet(
+					StringBundler.concat(
+						ObjectRelationship.class.getSimpleName(), "#",
+						objectDefinition.getExternalReferenceCode()),
```

Comment:
Hi @MarinhoFeliphe, totally agree with you on this. But I haven't been able to find a common term to use. We have the relation in the for provider, but not in the field values provider. Is there an easy way to retrive the relation with the field? Something that you would suggest.

Really thankful for your time!

Reply:
Btw, to start I only allowed 1 to many relationships, the many to many would be done in a second task (unless we see it's easy to do)

Reply:
@jkappler, sorry for the late reply,

> Is there an easy way to retrive the relation with the field? 

I'll provide here as soon as possible.

> the many to many would be done in a second task

makes sense to me, thank you!

Reply:
@jkappler, I believe we can define the namespace this way below:

````
StringBundler.concat(
    ObjectRelationship.class.getSimpleName(), "#",
    "#", objectDefinition.getName(), "#", objectRelationship.getName())
````

I avoid use definition's ERC in this case because it's mutable and I'm not sure if it will be a problem when the user change it, what do you think?

> Is there an easy way to retrive the relation with the field?

Yes, see this method: [ObjectRelationshipLocalService.fetchObjectRelationshipByObjectFieldId2](https://github.com/liferay/liferay-portal/blob/6aecb9985ca0972a7280dda175852d0838a6ee92/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java#L432-L437)

Reply:
I will check the method that you mentioned. Thanks again!